### ISSUE TITLE
Web: Canvas dimensions not correct initially.

### ISSUE THREAD

**Issue by bspot** [rust-windowing/winit#2524]
On the web, winit accounts for `devicePixelRatio` by setting both the canvas' `width/height` and applying CSS rules `width: ... / height: ...`.

However, the CSS rules are not applied initially, so the canvas is sized incorrectly.

After 
  * changing the `devicePixelRation` by zooming in or out, or
  * changing the window size
the CSS rules are then applied.

I believe the CSS rules should also be applied directly after the canvas is created.

**Comment by jroddev** [comment]
I came across this issue today with bevy/wgpu and it's causing the application to crash.
Running in Chromium based browsers (Vanadium, Brave) on a Pixel 7 with a devicePixelRatio of 2.62 the canvas exceeds the maximum texture limit in wgpu.
- *Chromium on Desktop Linux is fine.* 
- *Firefox on the same Android device also works.*

```
Caused by:
    In Device::create_texture
      note: label = `main_texture_a`
    Dimension Y value 4888 exceeds the limit of 4096
```
![image](https://user-images.githubusercontent.com/9654410/233827045-4bd46ff2-9ea4-4e72-975d-72088c6e864e.png)


The container is 980 x 1529 and the canvas is 2572 x 4015 (2.62 times the size it should be). The style width/height is correct.
![image](https://user-images.githubusercontent.com/9654410/233827396-b87b904c-8e0e-4fef-9ffa-7e2a39336388.png)



If I pad out other components to bring the canvas size under the limit then everything works.
![image](https://user-images.githubusercontent.com/9654410/233827049-282bcde8-6610-4f8e-99c4-7786682d6edf.png)

Bevy has a scale factor override but it doesn't look to be used by winit
```
pub fn scale_factor() -> f64 {
  let window = web_sys::window().expect("Failed to obtain window");
  window.device_pixel_ratio()
}
```
Can this be overridden?

**Comment by daxpedda** [comment]
I'm not 100% sure but I believe this was actually related to #2778. The initial canvas dimensions supplied by Winit were set though, they were just weird because of the relationship described in #2778.

In any case, it's definitely fixed in #2849.
See also https://github.com/gfx-rs/wgpu/pull/3690.
